### DISCUSSION TITLE
MenuItem InputGesture

### DISCUSSION THREAD

**Discussion by StefanKoell** [AvaloniaUI/Avalonia#13140] [General]
Hi,

it seems that the InputGesture property of the MenuItem class only allows you to set one KeyGesture - which is then rendered right-aligned in the menu item.

I plan to use "chord" keyboard shortcuts (like Ctrl+M, O or Ctrl+M, Ctrl+O) and handle the gestures/keyboard shortcut completely myself. Unfortunately it's not possible to provide an arbitrary string as an alternative to the InputGesture.

This PR shows the journey to the KeyGesture approach:
https://github.com/AvaloniaUI/Avalonia/pull/3602

The way it is implemented right now, is really limiting and doesn't allow a way to show chord keyboard shortcuts.

Any suggestions?

**Comment by timunie** [comment]
idea: use the Tag property or an attached property. And a custom style for MenuItem displaying that string.

**Reply by StefanKoell** [reply]
I want to avoid custom style/template for the menu item to avoid getting updates with new versions. I thought of inheriting from KeyGesture but unfortunately this is a sealed class. Wouldn't it make sense to unseal it to allow these special scenarios?
